test/cookbooks/cerner_splunk_ingredient_test/recipes/splunk_app.rb

```diff
@@ -0,0 +1,27 @@
+splunk_app_custom 'test_app' do
+  configs(proc do
+    splunk_conf 'testing.conf' do
+      config(
+        debug: {
+          banana: 'yellow'
+        }
+      )
+    end
+  end)
+
+  files(proc do |app_path|
+    file Pathname.new(app_path).join('plain_file.txt').to_s do
```

Comment:
The files property calls a proc and gives the app path as a parameter, and that's it. Consumers are intended to use any number of file and directory resources inside this proc to create non-config files. It doesn't make any assumptions about the contents or try to enforce it, so you could create the data/ui/views directory in the same proc and then create that file.